### PR TITLE
Added queries to fetch the list of supported languages and countries, resolves #16.

### LINKED ISSUE #16: Add query to request languages and calendars

### PR DESCRIPTION
**Pull request by vinczebalazs**

### CODE REVIEW

File: ApiClient/Source/Localization/Models/Protocols/Localization.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import Foundation
+
+public protocol Localization {
```
Comment (dmfs) [inline ApiClient/Source/Localization/Models/Protocols/Localization.swift]: Is the name really correct like this? Sounds like this should be called `Country`.

Comment (vinczebalazs) [reply ApiClient/Source/Localization/Models/Protocols/Localization.swift]: This protocol is used both by Language and Country, thus the name Localization.

File: ApiClient/Source/Localization/Queries/CountryQuery.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import Foundation
+import Alamofire
+
+public class CountryQuery: Query {
+    public typealias Result = [Localization]
+    
+    public let url: URL = URL(string:"https://api.schedjoules.com/countries")!
+    public let method: HTTPMethod = .get
+    public let parameters: Parameters = [:]
+    public let headers: HTTPHeaders = ["Accept" : "application/json", "Content-Type" : "application/json"]
```
Comment (dmfs) [inline ApiClient/Source/Localization/Queries/CountryQuery.swift]: The `Content-Type` header should not be present on a `GET` request. This reader identifies the media type of the request content, but there is none for a `GET`. So technically it's nonsense to specify a `Content-Type` of a request which doesn't carry any "content". I guess this applies to the other `Query`s too. I must have missed that during the reviews. Please open an issue.

File: ApiClient/Source/Localization/Queries/CountryQuery.swift [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import Foundation
+import Alamofire
+
+public class CountryQuery: Query {
```
Comment (dmfs) [inline ApiClient/Source/Localization/Queries/CountryQuery.swift]: I think it's better to be a bit more specific with the name. How about `SupportedCountriesQuery`?.

**Comment by dmfs** [comment]
@vinczebalazs please rebase and fix conflicts

File: ApiClient/Source/Localization/Models/Protocols/Language.swift [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+//  Created by Balazs Vincze on 2018. 04. 18..
+//  Copyright © 2018. SchedJoules. All rights reserved.
+//
+
+import Foundation
+
+public protocol Language {
+    var name: String { get }
+    var code: String { get }
+    var icon: URL? { get }
```
Comment (dmfs) [inline ApiClient/Source/Localization/Models/Protocols/Language.swift]: Does the API say languages may have an icon?

File: ApiClient/Source/Localization/Queries/SupportedCountriesQuery.swift [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import Foundation
+import Alamofire
+
+public class SupportedCountriesQuery: Query {
+    public typealias Result = [Country]
+    
+    public let url: URL = URL(string:"https://api.schedjoules.com/countries")!
+    public let method: HTTPMethod = .get
+    public let parameters: Parameters = [:]
+    public let headers: HTTPHeaders = ["Accept" : "application/json", "Content-Type" : "application/json"]
```
Comment (dmfs) [inline ApiClient/Source/Localization/Queries/SupportedCountriesQuery.swift]: `Content-type` still in here?

File: ApiClient/Source/Localization/Queries/SupportedLanguagesQuery.swift [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+import Foundation
+import Alamofire
+
+public class SupportedLanguagesQuery: Query {
+    public typealias Result = [Language]
+    
+    public let url: URL = URL(string:"https://api.schedjoules.com/languages")!
+    public let method: HTTPMethod = .get
+    public let parameters: Parameters = [:]
+    public let headers: HTTPHeaders = ["Accept" : "application/json", "Content-Type" : "application/json"]
```
Comment (dmfs) [inline ApiClient/Source/Localization/Queries/SupportedLanguagesQuery.swift]: `Content-Type` still in here as well?

**Pull request merged**